src/Control/Monad/Reader/Trans.purs

```diff
@@ -96,3 +100,12 @@ instance monadWriterReaderT :: (Monad m, MonadWriter w m) => MonadWriter w (Read
 instance distributiveReaderT :: (Distributive g) => Distributive (ReaderT e g) where
   distribute a = ReaderT \e -> collect (flip runReaderT e) a
   collect f = distribute <<< map f
+
+instance monadRecReaderT :: (MonadRec m) => MonadRec (ReaderT r m) where
+  tailRecM k a = ReaderT \r -> tailRecM k' (Tuple a r)
```

Comment:
I don't think you need to tuple up the state here. You can just pass in `r` as it doesn't change.